### PR TITLE
Feature judging

### PR DESCRIPTION
**Pull request by Abhiman2211**
## Description
Allows 4 new functions with relation to scoring. You can now add, subtract, set or reset a team's score. Don't approve as I am still working on adding tests. Feedback would be appreciated though.

## Checks
- [] No errors when running the tests
- [] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -66,6 +71,23 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+     *
+     * @param score the string representation of the score under check.
+     * @throws ParseException if the score is not a valid integer.
+     */
+    public static void validateScore(String score) throws ParseException {
+        if (score.equals("")) {
+            throw new ParseException(MESSAGE_INVALID_COMMAND_FORMAT);
+        }
+        try {
+            int scoreValue = Integer.parseInt(score);
```
Comment (brianyenna) [inline src/main/java/seedu/address/commons/util/StringUtil.java]: Just to check, score can be negative as well right

Comment (Abhiman2211) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: nope it cant
In the score class we set a restriction for it to be between 0 and 100

File: src/main/java/seedu/address/logic/commands/scorecommand/AddScoreCommand.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        try {
+            teamToScore = model.getTeam(id);
+        } catch (AlfredException ae) {
+            throw new CommandException(MESSAGE_INVALID_TEAM_DISPLAYED_INDEX);
+        }
+
+        try {
+            model.addTeamScore(teamToScore, score);
+        } catch (AlfredException e) {
```
Comment (brianyenna) [inline src/main/java/seedu/address/logic/commands/scorecommand/AddScoreCommand.java]: this is a little nitpicky, but AlfredException is given the variable e here, but is given the variable ae above. LOL

Comment (Abhiman2211) [reply src/main/java/seedu/address/logic/commands/scorecommand/AddScoreCommand.java]: sure i'll change it

File: src/main/java/seedu/address/logic/commands/scorecommand/SubtractScoreCommand.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    public static final String MESSAGE_SCORE_TEAM_SUCCESS = "Subtracted %1$s points from %2$s's score";
+    public static final String COMMAND_WORD = "score sub";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": subtracts the specified score from the specified team's current score. "
+            + "If the team's new score goes below " + Score.MIN_SCORE + " it will be set to " + Score.MIN_SCORE + "\n"
+            + "Format: " + COMMAND_WORD + " teamID score \n"
+            + "For example: " + COMMAND_WORD + " T-5 15";
+
+    public SubtractScoreCommand(Id teamId, Score score) {
```
Comment (brianyenna) [inline src/main/java/seedu/address/logic/commands/scorecommand/SubtractScoreCommand.java]: hey what if score is given a negative value here. Would this then become addition

Comment (Abhiman2211) [reply src/main/java/seedu/address/logic/commands/scorecommand/SubtractScoreCommand.java]: the score can't be negative, so this wont happen
in fact that is why there is a subtract and add operation

**Pull request merged**